### PR TITLE
Release version 1.0.0

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Version 1.0 major release

## Checklist:

- [x] I agree with the terms of the [Contributor Agreement](https://github.com/glotzerlab/synced_collections/blob/main/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/synced_collections/blob/main/contributors.yaml).
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.
- [x] I have updated the [changelog](https://github.com/glotzerlab/synced_collections/blob/main/changelog.txt) and added any related issue and pull request numbers for future reference.

### CODE REVIEW

**Comment by vyasr** [comment]
More information on the changes to the publish workflow's authentication, see https://docs.pypi.org/trusted-publishers/.

**Comment by vyasr** [comment]
Would also recommend that reviewers have a look at https://github.com/glotzerlab/synced_collections/pull/7, which sets up automated Github Releases. I merged that quickly so that I could verify that the deployment workflows for PyPI (or at least TestPyPI) worked correctly in this PR, but I am happy to edit the new GH release workflow as well in this PR.

**Review by bdice — CHANGES_REQUESTED** [review]
Blocking on this until some issues in #7 are addressed. I wish I could have reviewed that PR before merge... but I'll leave some comments and file a follow-up there.

**Comment by bdice** [comment]
Opened #9. I'd like to merge that before this PR is merged.

**Comment by vyasr** [comment]
> Opened #9. I'd like to merge that before this PR is merged.

Thanks @bdice! Your PR is merged now. Let me know if you'd like to see anything else. No rush, I would like at least one more approval before I merge this PR in any case.

**Review by bdice — APPROVED** [review]
With #9 merged, I approve this.

**Review by bdice — CHANGES_REQUESTED** [review]
@vyasr I thought some more and think I want to block on this. Before we release 1.0.0 here, it might be good to try the packaging in practice. I'll create a test branch in signac that installs this from git and removes the existing `synced_collection` code.

**Comment by bdice** [comment]
I opened these PRs. Looks like things are passing against the `main` branch here, so we can merge this PR and release 1.0.0 before or after merging the downstream PRs.
- https://github.com/glotzerlab/signac/pull/924
- https://github.com/glotzerlab/signac-docs/pull/189

**Review by bdice — APPROVED** [review]

File: changelog.txt
Comment (b-butler) [inline changelog.txt]: Will need to be modified

**Review by b-butler — APPROVED** [review]
I am happy with this. We just need to change the release date.

File: changelog.txt
Comment (vyasr) [inline changelog.txt]:
```suggestion
[1.0.0] -- 2023-05-17
```

**Pull request merged**
